### PR TITLE
CI: port cygwin job from appveyor to azure-pipelines

### PR DESCRIPTION
**Pull request by lazka**

### CODE REVIEW

**Comment by jpakkane** [comment]
LGTM. @nirbheek, what do you think?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -51,3 +51,46 @@ jobs:
...
+        gcc-objc,^
+        gobject-introspection,^
+        libboost-devel,^
+        libglib2.0-devel,^
+        libgtk3-devel,^
+        ninja,^
+        python3-pip,^
+        vala,^
+        zlib-devel,^
+        git
```
Comment (jon-turney) [inline azure-pipelines.yml]: Please keep this list of packages in alphbetical order

File: azure-pipelines.yml [hunk 2]
```diff
@@ -51,3 +51,47 @@ jobs:
 
   steps:
   - template: ci/azure-steps.yml
+
+- job: cygwin
```
Comment (jon-turney) [inline azure-pipelines.yml]: Might make sense to name this something like `cygwin gccx64ninja` for consistency with the other jobs

**Comment by lazka** [comment]
done

**Pull request merged**